test/infrastructure/container/docker.go

```diff
@@ -0,0 +1,400 @@
+/*
```

Comment:
I like this module, this is the right direction!
What about:
- Adding an Interface type exposing the public methods defined in this file , `container.Interface`
- Adding a `container.docker` struct, implementing `container.Interface` (in practice the current code with `func (d *docker) methodName`

So
- We can implement a ``container.NewDockerClient() (container.Interfac, error) ` method that returns an instance of the `container.docker` type
- We can store the `dockerClient` at struct level, set it at create time, and share across all the calls
- (Most important) We are paving the way for injecting other container implementations, or a fake one for testing